codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/serialize/JsonSerializerGenerator.kt

```diff
@@ -49,11 +51,27 @@ import software.amazon.smithy.rust.codegen.util.hasTrait
 import software.amazon.smithy.rust.codegen.util.inputShape
 import software.amazon.smithy.rust.codegen.util.outputShape
 
+/**
+ * Class describing a JSON section that can be used in a customization.
+ */
+sealed class JsonSection(name: String) : Section(name) {
+    /**
+     * Write the customization just before `object.finish()`.
+     */
+    data class FinalizeObject(val structureShape: StructureShape) : JsonSection("FinalizeObject")
```

Comment:
pass the name of the `object` as a string so that implementors don't need to guess / hardcode it (they can inspect the section then use that during codegen)